src/datanode/src/instance.rs

```diff
@@ -141,8 +144,12 @@ impl Instance {
                         Some(catalog as TableIdProviderRef),
                     )
                 } else {
+                    let engine_manager = Arc::new(MemoryTableEngineManager::new(
+                        MITO_ENGINE,
+                        table_engine.clone(),
+                    ));
```

Comment:
Since we already have the `name` method in TableEngine trait:

```suggestion
                    let engine_manager = Arc::new(MemoryTableEngineManager::new());
                    engine_manager.register_engine(table_engine.name(), table_engine.clone());
```